components/builder-worker/src/runner/docker.rs

```diff
@@ -0,0 +1,197 @@
+        self.teardown_dockerd(dockerd)?;
+        exit_status
+    }
+
+    fn run_export(&self, log_pipe: &mut LogPipe) -> Result<ExitStatus> {
+        let sock = self.dockerd_sock();
+
+        let mut cmd = Command::new(&*DOCKER_EXPORTER_PROGRAM);
+        cmd.current_dir(self.workspace.root());
+        cmd.arg("--base-pkgs-url");
+        cmd.arg(&self.bldr_url);
+        cmd.arg("--url");
+        cmd.arg(&self.bldr_url);
+        cmd.arg("--tag-latest");
+        cmd.arg("--tag-version");
+        cmd.arg("--tag-version-release");
+        // TODO fn: enable push once passwords are gettable
+        // cmd.arg("--push-image");
+        // cmd.arg("--username");
+        // cmd.arg(username);
+        // cmd.arg("--password");
+        // cmd.arg(password);
+        // TODO fn: cleanup image once pushing is rolling
+        // cmd.arg("--rm-image");
+        cmd.arg(self.workspace.last_built()?.path); // Locally built artifact
+        debug!("building docker export command, cmd={:?}", &cmd);
+
+        cmd.env_clear();
+        if let Some(_) = env::var_os(RUNNER_DEBUG_ENVVAR) {
+            cmd.env("RUST_LOG", "debug");
+        }
+        cmd.env(NONINTERACTIVE_ENVVAR, "true"); // Disables progress bars
+        cmd.env("TERM", "xterm-256color"); // Emits ANSI color codes
+        debug!(
+            "setting docker export command env, {}={}",
+            DOCKER_HOST_ENVVAR,
+            sock
+        );
+        cmd.env(DOCKER_HOST_ENVVAR, sock); // Use the job-specific `dockerd`
+        cmd.stdout(Stdio::piped());
+        cmd.stderr(Stdio::piped());
+
+        debug!("spawning docker export command");
+        let mut child = cmd.spawn()?;
+        log_pipe.pipe(&mut child)?;
+        let exit_status = child.wait()?;
+        debug!("completed docker export command, status={:?}", exit_status);
+        Ok(exit_status)
+    }
+
+    fn spawn_dockerd(&self) -> Result<Child> {
+        let root = self.dockerd_path();
+        let env_path = &*DOCKERD_PROGRAM.parent().expect(
+            "Dockerd parent directory exists",
+        );
+        let daemon_json = root.join("etc/daemon.json");
+        fs::create_dir_all(daemon_json.parent().expect(
+            "Daemon JSON parent directory exists",
+        ))?;
+        {
+            let mut f = File::create(&daemon_json)?;
+            f.write_all(b"{}")?;
+        }
+
+        let mut cmd = Command::new(&*DOCKERD_PROGRAM);
+        if let Some(_) = env::var_os(RUNNER_DEBUG_ENVVAR) {
+            cmd.arg("-D");
+        }
+        cmd.arg("-H");
+        cmd.arg(self.dockerd_sock());
+        cmd.arg("--pidfile");
+        cmd.arg(root.join("var/run/docker.pid"));
+        cmd.arg("--data-root");
+        cmd.arg(root.join("var/lib/docker"));
+        cmd.arg("--exec-root");
+        cmd.arg(root.join("var/run/docker"));
+        cmd.arg("--config-file");
+        cmd.arg(daemon_json);
+        // TODO fn: Hard-coding this feels wrong. I'd like the {{pkg.svc_run_group}} for this
```

Comment:
I would think you'd want to spawn it as the same user/group as the worker itself or perhaps it's a configuration on the worker for what user you'd like to run Docker as?

Reply:
It sets the group ownership of the `docker.sock` which determines access. I'm hopeful if I plumb though the runtime variable this should work--even though that group is most likely `root`.